shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -673,6 +674,28 @@ - (BOOL)runWithEntrypoint:(NSString*)entrypoint {
     [engine onVSync:baton];
   };
 
+  NSString* bundlePath =
+      [[NSBundle bundleWithURL:[NSBundle.mainBundle.privateFrameworksURL
+                                   URLByAppendingPathComponent:@"App.framework"]] bundlePath];
+  bundlePath = [bundlePath stringByAppendingString:@"/App"];
+  flutterArguments.app_path = bundlePath.UTF8String;
```

Comment:
What's the lifetime of this string?  How long does flutterArguments expect this char* to be alive?

Reply:
Should be fine to collect after the call to initialize the Flutter engine via the Embedder API.